### PR TITLE
fix potential fanout in customer metrics model

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**Are you a current Fivetran customer?** 

internal 

**What change(s) does this PR introduce?** 

- adds flow, campaign, and variation id to the joins in the customer metrics model (they are a part of the PK grain of this model, so we should definitely have them here!)
- adds coalesces to the above joins
- updates to use newer version of shopify package
  - add new seed files
  - rename seeds to not use the `identifier` variable  (integration tests are not working well with union_data) 
  -  the customer enhanced model now refers to the new `shopify__customer_emails` model instead of recreating it in `int__shopify_customer_rollup` (deleted)
- adds incremental runs to buildkite
- updates changelog
- updates README to our current template
- comments out identifiers in integration_tests, as they are un-needed given the (new) seed names
  - adds alias for `order` seed and uses that as an alias

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [x] Yes, Issue https://github.com/fivetran/dbt_shopify_holistic_reporting/issues/14, https://github.com/fivetran/dbt_shopify_holistic_reporting/issues/12, https://github.com/fivetran/dbt_shopify_holistic_reporting/issues/10
- [ ] No 

**How did you test the PR changes?** 

 

- [x] Buildkite  
- [x] Local (please provide additional testing details below)

> - adds flow, campaign, and variation id to the joins in the customer metrics model (they are a part of the PK grain of this model, so we should definitely have them here!)
> - adds coalesces to the above joins

customer validated the above

> - updates to use newer version of shopify package
>   - refers to the new `shopify__customer_emails` model instead of recreating it in `int__shopify_customer_rollup` (deleted)

Validated below (and came across an issue in the shopify transform package for which I've opened https://github.com/fivetran/dbt_shopify/pull/58)
image

^ this actually originally didn't tie out.  but it turns out the new version is more accurate, because previously we didn't filter out:
- `void` transactions
- `authorization` transactions (often charges will have duplicate transaction records where one of them is just a hold on the card)
- gift card payments (edeeming a giftcard does not introduce new revenue and thus shouldn't be included)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🌞 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Review by fivetran-avinash — CHANGES_REQUESTED** [review]
Hey @fivetran-jamie! Just a few issues to resolve here! My major one was I wasn't able to replicate your empty data set for the new vs. old model.  Happy to debug it today if you're not able to replicate and I can show you the steps I took to replicate. Maybe I missed a step somewhere!

File: models/shopify_holistic_reporting__customer_enhanced.sql
Comment (fivetran-avinash) [inline models/shopify_holistic_reporting__customer_enhanced.sql:4]: See Height ticket regarding inability to reproduce new data set.

Comment (fivetran-avinash) [reply models/shopify_holistic_reporting__customer_enhanced.sql:4]: Able to reproduce!

File: .buildkite/scripts/run_models.sh
Comment (fivetran-jamie) [reply .buildkite/scripts/run_models.sh:23]: https://github.com/fivetran/dbt_shopify_holistic_reporting/blob/main/models/shopify_holistic_reporting__orders_attribution.sql

File: CHANGELOG.md
Comment (fivetran-jamie) [reply CHANGELOG.md:19]: hm yeah i was wondering tho if it's duplicative to summarize the shopify release here since we are already linking to the shopify release notes (so users can read the summary there). what do you think?

File: CHANGELOG.md
Comment (fivetran-jamie) [reply CHANGELOG.md:20]: sam comment as above -- is it duplicative when we are linking to the individual package release notes?

File: README.md
Comment (fivetran-jamie) [reply README.md:85]: how come?

Comment (fivetran-avinash) [reply README.md:85]: oh wait my bad i didn't see klaviyo_source and shopify_source were also deleted. Ignore.

File: README.md
Comment (fivetran-avinash) [reply README.md:88]: Nope, ignore this.

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md]: Yeah I'm not sure. I'm realizing there aren't that many shopify holistic customers at this point so maybe it's not needed. I'll leave it to your discretion.

File: integration_tests/dbt_project.yml
Comment (fivetran-jamie) [reply integration_tests/dbt_project.yml:14]: they are indeed meant to be commented out! i can remove them if they're confusing, just figured i'd leave examples in here

IMO changes to seed files don't need to go in the CHANGELOG since they don't affect users at all...but are there reasons that users would want to be aware of these sorts of changes? 🤔

File: models/shopify_holistic_reporting__customer_enhanced.sql
Comment (fivetran-jamie) [reply models/shopify_holistic_reporting__customer_enhanced.sql:30]: > looks like last_updated_at is removed from this CTE so is it needed?

so are you saying we should remove it from the `except` list? not sure exactly what ya mean 

> Also seems we're relying on first_name and last_name now from shopify_customers instead of full_name

so `full_name` is a field we used to create ourselves in the [int model](https://github.com/fivetran/dbt_shopify_holistic_reporting/blob/main/models/intermediate/int__shopify_customer_rollup.sql#L19) this PR deletes, so i just moved _where_ we perform the concatenation

Comment (fivetran-avinash) [reply models/shopify_holistic_reporting__customer_enhanced.sql:30]: > so are you saying we should remove it from the except list? not sure exactly what ya mean

I think I'm looking at the renaming of `last_updated_at` to `updated_timetamp`, so maybe it doesn't have to be deleted but just renamed to `updated_timestamp` (If the intent is to bring in all the columns not already added into this model)?  Or does the `relation_alias` handle this by looking directly within the cte? 

> so full_name is a field we used to create ourselves in the [int model](https://github.com/fivetran/dbt_shopify_holistic_reporting/blob/main/models/intermediate/int__shopify_customer_rollup.sql#L19) this PR deletes, so i just moved where we perform the concatenation

Oh sorry I meant in reference to the except statement in the `dbt_utils.star` function, it seems like we'd bring in `first_name` and `last_name` but we've already produced `full_name`. However it looks like that was the earlier behavior so it's probably fine the way it is.

Comment (fivetran-jamie) [reply models/shopify_holistic_reporting__customer_enhanced.sql:30]: ohhhhhh yes i will add `updated_timetamp` to the except list! i should also add `created_timestamp`. the timestamp fields got renamed in a recent version of shopify, so they got all jumbled up together

i am actually going to revert line 23 though just for consistency (`last_updated_at` is also just calculated in a more reader-friendly way than `updated_timestamp`) 

> Oh sorry I meant in reference to the except statement in the dbt_utils.star function, it seems like we'd bring in first_name and last_name but we've already produced full_name. However it looks like that was the earlier behavior so it's probably fine the way it is.

Ahh sorry I see what you mean!  i'll replace `full_name` in the except list with `first_name, last_name`

Comment (fivetran-avinash) [reply models/shopify_holistic_reporting__customer_enhanced.sql]: Sounds great! Thanks for the walkthrough. All the new changes look good!

**Review by fivetran-avinash — APPROVED** [review]
PR approved! Gotta get pesky Databricks to pass but after that free to release/merge.

**Pull request merged**
